lib/daru/index/multi_index.rb

```diff
@@ -14,16 +14,30 @@ def levels
       @levels.map(&:keys)
     end
 
+    # names and levels should be of same size. If size of Array `name` is less
+    # or greater than size of array `levels` then raises `SizeError`.
+    # If user don't want to put name for particular level then user must put
+    # empty string in that index of Array `name`.
+    # For example there is multi_index of 3 levels and user don't want to name
+    # level 0, then do mulit_index.name = ['', 'level1_name1', 'level2_name']
     def initialize opts={}
       labels = opts[:labels]
       levels = opts[:levels]
+      names = opts[:name]
 
       raise ArgumentError, 'Must specify both labels and levels' unless labels && levels
       raise ArgumentError, 'Labels and levels should be same size' if labels.size != levels.size
       raise ArgumentError, 'Incorrect labels and levels' if incorrect_fields?(labels, levels)
+      validate_name names, levels unless names.nil?
 
       @labels = labels
       @levels = levels.map { |e| e.map.with_index.to_h }
+      @name = names
```

Comment:
thanks , I have edited the PR accordingly in this commit https://github.com/SciRuby/daru/pull/323/commits/5955197912cd83223c9f790c9e4264389a0348e4